backend/src/elasticsearch.ts

```diff
@@ -0,0 +1,67 @@
+import { Client, ClientOptions } from '@elastic/elasticsearch';
+import loggerStream from './logger';
+
+// Configuration du client pour utiliser Undici par défaut
+const config: ClientOptions = {
+  node: 'http://elasticsearch:9200',
+  requestTimeout: 30000, // Timeout de 30 secondes pour les requêtes
+  maxRetries: 3,         // Nombre maximum de tentatives en cas d'échec
+  // Laisser Undici gérer les options de connexion/socket par défaut.
+  // Si des ajustements fins sont nécessaires plus tard, nous pourrons ajouter un objet `agent`
+  // avec des options spécifiques à Undici (ex: connections, pipelining).
+};
+
+// Création du client
+let esClientInstance: Client | null = null;
```

Comment:
'Client' is an 'error' type that acts as 'any' and overrides all other types in this union type  @typescript-eslint/no-redundant-type-constituents

```suggestion
let esClientInstance: Client = null;
```

Reply:
Pris en compte !